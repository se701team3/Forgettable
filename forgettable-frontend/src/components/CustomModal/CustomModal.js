import React from 'react';
import classes from './CustomModal.module.css';
import {Modal} from '@mui/material';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import CustomButton from '../CustomButton/CustomButton';

function CustomModal(props) {
  const {open, onClose, hasCancel, hasConfirm, onConfirm} = props;

  return (
    <Modal
      open={open}
      onClose={onClose}
    >
      <div className={classnames(classes.Container, props.className)}>
        <div>{props.children}</div>
        <div className={classes.Footer}>
          {hasCancel && <CustomButton btnText={'Cancel'} onClick={onClose}/>}
          {hasConfirm && <CustomButton btnText={'Confirm'} onClick={onConfirm}/>}
        </div>
      </div>
    </Modal>);
}

CustomModal.propTypes = {
  open: PropTypes.bool.isRequired,
  onClose: PropTypes.func.isRequired,
  hasCancel: PropTypes.bool,
  hasConfirm: PropTypes.bool,
  onConfirm: PropTypes.func,
};

export default CustomModal;
