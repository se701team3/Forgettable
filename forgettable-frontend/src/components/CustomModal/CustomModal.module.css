.Container{
  width: fit-content;
  height: fit-content;
  min-width: 525px;
  min-height: 300px;
  max-height: 85%;
  max-width: 65%;
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  background-color: white;
  border-radius: 13px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  padding: 20px;
  padding-bottom: 90px;
  overflow-y: scroll;
}

.Container::-webkit-scrollbar { 
  display: none;  /* Safari and Chrome */
}

.Footer{
  display: flex;
  flex-direction: row;
  gap: 20px;
  position: absolute;
  bottom: 30px;
}