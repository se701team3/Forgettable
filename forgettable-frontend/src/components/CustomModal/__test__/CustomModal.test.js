import TestRenderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';
import React from 'react';
import CustomModal from '../CustomModal';
import {render, screen} from '@testing-library/react';

let modalOpen = true;
const handleModalClose = () => {
  modalOpen = false;
};

it('renders CustomModal UI to test correct hierarchy', () => {
  const renderer = new ShallowRenderer();
  renderer.render(
      <CustomModal
        open={modalOpen}
        onClose={handleModalClose}
        hasCancel={true}
        hasConfirm={true}
        onConfirm={handleModalClose}
      >
        <h1>This is an example title</h1>
      </CustomModal>);

  const result = renderer.getRenderOutput();

  expect(result).toMatchSnapshot();
});

it('check if children components are passed in correctly', () => {
  render(
      <CustomModal
        open={true}
        onClose={handleModalClose}
        hasCancel={true}
        hasConfirm={true}
        onConfirm={handleModalClose}
      >
        <h1>Title</h1>
      </CustomModal>);

  const titleElement = screen.getByText(/title/i);

  expect(titleElement).toBeInTheDocument();
});
