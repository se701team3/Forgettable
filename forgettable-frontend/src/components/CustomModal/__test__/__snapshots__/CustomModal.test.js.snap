// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CustomModal UI to test correct hierarchy 1`] = `
<ForwardRef(Modal)
  onClose={[Function]}
  open={true}
>
  <div
    className="Container"
  >
    <div>
      <h1>
        This is an example title
      </h1>
    </div>
    <div
      className="Footer"
    >
      <CustomButton
        btnText="Cancel"
        onClick={[Function]}
      />
      <CustomButton
        btnText="Confirm"
        onClick={[Function]}
      />
    </div>
  </div>
</ForwardRef(Modal)>
`;
